src/API/users/PTO.js

```diff
@@ -0,0 +1,119 @@
+const logger = require("../../log/logger");
+const checkin = require("../../data/schema/checkin-schema");
+const users = require("../../data/schema/user-schema");
+const moment = require("moment");
+
+module.exports = function(req, res) {
+  if ("pin" in req.body && "day" in req.body) {
+    const { pin, day } = req.body;
+
+    if (isValidNumber(pin)) {
+      users.findOne({ pin: pin }, (err, item) => {
+        if (err) {
+          logger.warn(err);
+        }
+        if (item) {
+          let id = item._id;
+          let name = item.name;
+          const date = new Date();
+          if (!day) {
+            date.setDate(date.getDate() - 1);
+          }
+          const currentDate = `${date.toLocaleDateString()}`;
+          checkin.findOne(
+            { user_id: id, date: { $regex: date.toLocaleDateString() } },
+            (err, item) => {
+              if (err) {
+                logger.warn(err);
+              }
+              if (!item) {
+                try {
+                  const weekNumber = moment(
+                    moment().format("M/D/YYYY")
+                  ).isoWeek();
+                  checkin.find(
+                    { user_id: id, week: weekNumber - 1 },
+                    (err, items) => {
+                      if (err) {
+                        logger.warn(err);
+                      }
+                      let average = 0;
+                      if (items) {
+                        let totalMinutes = 0;
+                        items.map(item => {
+                          totalMinutes = totalMinutes + item.minutes;
+                        });
+                        average = Math.round(totalMinutes / items.length);
+                      }
+                      const currentDate = `${date.toLocaleDateString()} ${date.toLocaleTimeString()}`;
+                      const check = new checkin({
+                        user_id: id,
+                        pto: true,
+                        date: currentDate,
+                        minutes: average >= 0 ? average : 0,
+                        week: weekNumber
+                      });
+                      checkin.create(check, (err, item) => {
+                        if (err) logger.warn(err);
+                        if (item) {
+                          let resp = {
+                            info: `PTO successfully created, your penalization time is ${
+                              average >= 0 ? average : 0
+                            } minutes`,
+                            type: "success"
+                          };
+                          res.status(201).send(JSON.stringify(resp));
+                        } else {
+                          let resp = {
+                            info: `Something went wrong`,
```

Comment:
We should change this message to something more specific. Where also missing unit tests for this.

Reply:
Done